app/pages/resource/reservation-calendar/time-slots/TimeSlots.js

```diff
@@ -24,30 +27,85 @@ class TimeSlots extends Component {
     time: PropTypes.string,
   };
 
+  calculatePlaceholders(selectedDate, slots) {
+    const firstTimeSlots = slots
+      .map(timeslots => timeslots && timeslots[0])
+      .filter(value => !!value && value.end);
+
+    if (firstTimeSlots.length === 0) {
+      return {
+        mobilePlaceholderOffset: 0,
+        timeSlotPlaceholderSizes: Array(slots.length).fill(0),
+      };
+    }
+
+    const earliestTimeSlot = minBy(firstTimeSlots, timeSlot =>
+      moment(timeSlot.start).format('HHMM')
+    );
+    const dateForTimeComparison = { year: 2000, dayOfYear: 1 };
+    const earliestStart = moment(earliestTimeSlot.start).set(dateForTimeComparison);
+
+    const timeSlotPlaceholderSizes = slots.map((slot) => {
+      if (!slot[0] || !slot[0].end) {
+        return null;
+      }
+      const currentStart = moment(slot[0].start).set(dateForTimeComparison);
+      return currentStart.diff(earliestStart, 'minutes') / 30;
+    });
+
+    const selectedDateIndex = findIndex(
+      slots,
```

Comment:
```suggestion
      firstTimeSlots,
```

Reply:
Has to be done over `slots` since we are finding an index and `firstTimeSlots` is filtered for days when the unit is closed (no `end` field).

Reply:
Ah yes!